week-5-ci-and-production/w05e02-continuous-deployment/.github/workflows/front.yml

```diff
@@ -0,0 +1,38 @@
+name: FRONT – Stage Deploy
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - 'shopping-lister-front/**'
+
+jobs:
+  test_build_and_deploy:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Cache node modules
+        uses: actions/cache@v3
+        with:
+          path: '**/node_modules'
+          key: ${{ runner.os }}-modules-${{ hashFiles('**/yarn.lock') }}
+      - name: Install dependencies
+        run: npm ci
+        working-directory: ./shopping-lister-front
+      - name: Format code and check for errors
+        run: npm run lint:full
+        working-directory: ./shopping-lister-front
+      - name: Run tests
+        run: npm run test:unit
+        working-directory: ./shopping-lister-front
+      - name: Build project
+        run: npm run build
+        working-directory: ./shopping-lister-front
+      - name: Deploy
+        run: |
+          mkdir -p ~/shopping-lister-front-stage
+          cp -a . ~/shopping-lister-front-stage/
```

Comment:
Tutaj powinieneś skończyć @teziovsky ,

Wystawianie tego przez `pm2` nie ma sensu.
Z założenia, `front` to tzw. "statyczne assety" więc masz je w jakimś katalogu, który jest wpięty jako `www root` w jednej z konfiguracji Twojego serwera `http` na maszynie.

wiec sam proces przekopiowania wystarczy.
+
powinien być jeszcze krok `rm -rf` folderu aktualnego (prod) i dopiero kopia aktualnego na jego miejsce

Reply:
Dodatkowo zły katalog jest kopiowany !

Twoje statyczne assety są w katalogu `/dist`. więc nie kopiujesz tak jak w przypadku serwera - całego tego katalogu z kodem, tylko `/dist`

Reply:
No tak, tak faktycznie. Normalnie faktycznie bym przekopiował te pliki do folderu, który byłby połączony z `nginxem`, ale nie byłem pewien czy samo przekopiowanie plików Cię usatysfakcjonuje jeżeli chodzi o wykonanie zadania!
Poprawiłem to w nowym commicie 👍